bot.py

```diff
@@ -4,35 +4,20 @@
 from hanabi import Board, Clue, Discard, Play
 
 class BaseBot():
-    def __init__(self):
-        pass
+    def __init__(self, board: Board, index: int):
+        self.board = board
```

Comment:
I made this change with the idea that bots should eventually be stateful. 

With the updated information system board is aware of "Explicit" information (e.g. clue touched a card) and "Direct Implicit" information (e.g. clue did not touch a card), however "Indirect Implicit" information (e.g. play clue vs save clue, prompt, finesse) is much more difficult to capture in the board. 

If a bot wants to consider "Indirect Implicit" information, then it needs to keep track of who said what and when. That was the idea behind stateful bots and the 'Listen' method.

Reply:
With `self.turns`, it should be possible to read from that instead. Not sure if that's preferable though

Reply:
I've gone back to the original pattern here. I agree with your point about testing hypotheticals.

However, `self.turns` is not enough of a history for the purposes of identifying play/save clues etc. because we don't store the starting deal so we can't re-create the state of the board purely from this history.